src/wp-includes/customize/class-wp-customize-custom-css-setting.php

```diff
@@ -163,8 +163,16 @@ public function validate( $value ) {
 
 		$validity = new WP_Error();
 
-		if ( preg_match( '#</?\w+#', $css ) ) {
-			$validity->add( 'illegal_markup', __( 'Markup is not allowed in CSS.' ) );
+		/**
+		 * Check for a closing STYLE tag inside the CSS.
+		 *
+		 * STYLE tags are processed using the "generic raw text parsing algorithm." They contain
+		 * raw text up until a matching closing tag.
+		 *
+		 * @see https://html.spec.whatwg.org/multipage/parsing.html#generic-raw-text-element-parsing-algorithm
+		 */
+		if ( stripos( $css, '</style' ) ) {
```

Comment:
This is missing the `false` check, right? At the moment, it won't be marked as invalid if the CSS starts with `</style` because `stripos()` will return `0`.

```suggestion
		if ( false !== stripos( $css, '</style' ) ) {
```